### PR TITLE
[FLOSS H3] csvstorage: fix silent failure on meta-key insertion

### LINKED ISSUE #4586: [FLOSS] csvstorage plugin fails silently on meta-set
As discussed in #4555
## Steps to Reproduce the Problem
```sh
$ kdb mount config.csv user:/tests/csv csvstorage  "header=colname,columns/names/#0=col0Name,columns/names/#1=col1Name"
# RET: 0
$ sudo kdb meta-set system:/tests/hosts somevalue a
# RET: 0  
$ sudo kdb meta-get system:/tests/hosts somevalue
Key not found
# RET: 1
```
## Expected Result

Should print an error on `sudo kdb meta-set system:/tests/hosts somevalue a` and return a non 0 exit code

## System Information
Operating System:
Arch Linux, Kernel 5.19.13-arch1-1

kdb and plugins version
```sh
# kdb --version
KDB_VERSION: 0.9.11
SO_VERSION: 5
```

## Relevant Source Code
csvstorage ignores all metadata other than type

https://github.com/ElektraInitiative/libelektra/blob/cf4b6f303432dbfdcb9cc222c84f41045962b211/src/plugins/csvstorage/csvstorage.c#L731

### PR DESCRIPTION
**Pull request by Bujuhu**
fixes #4586

## Basics

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation (see [Documentation Guidelines](https://www.libelektra.org/devgettingstarted/documentation))
- [ ] I fixed all affected decisions (see [Decision Process](https://www.libelektra.org/decisions/decision-process))
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://www.libelektra.org/devgettingstarted/coding))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- [ ] Add the "work in progress" label if you do not want the PR to be reviewed yet.
- [ ] Add the "ready to merge" label **if the basics are fulfilled** and no further pushes are planned by you.

### CODE REVIEW

**Comment by Dynamichost96** [comment]
Documentation is introductory, concise, good to read and describes everything what the PR does
CI fails due to out of memory error I assume this should not be related to the code changed!

Examples are well chosen and understandable
N/A

Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
Code is conforming to the Guidelines

APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
N/A

Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)
Code is consistent to the Guidelines

File: src/plugins/csvstorage/testmod_csvstorage.c [hunk 1]
```diff
@@ -84,6 +84,20 @@ static void testwriteinvalidheader (const char * file)
...
+static void testwriteInvalidMetaMustFail (const char * file)
+{
+	Key * parentKey = keyNew ("user:/tests/csvstorage", KEY_VALUE, srcdir_file (file), KEY_END);
+	KeySet * conf = ksNew (20, keyNew ("system:/delimiter", KEY_VALUE, ";", KEY_END),
+			       keyNew ("system:/header", KEY_VALUE, "colname", KEY_END), KS_END);
+	KeySet * ks = ksNew (1, keyNew ("user:/tests/csvstorage", KEY_VALUE, "asdf", KEY_META, "asdf", "asdf", KEY_END), KS_END);
+	PLUGIN_OPEN ("csvstorage");
+	succeed_if (plugin->kdbSet (plugin, ks, parentKey) == ELEKTRA_PLUGIN_STATUS_ERROR,
+		    "kdbSet did not error on invalid meta key insertion");
+	ksDel (ks);
```
Comment (flo91) [inline src/plugins/csvstorage/testmod_csvstorage.c:96]: Please also delete the `conf` keyset.
```suggestion
	ksDel (conf);
	ksDel (ks);
```

**Review by flo91 — CHANGES_REQUESTED** [review]
~Please also delete the `conf` keyset.~

File: src/plugins/csvstorage/testmod_csvstorage.c [hunk 2]
```diff
@@ -84,6 +84,21 @@ static void testwriteinvalidheader (const char * file)
...
+static void testwriteInvalidMetaMustFail (const char * file)
+{
+	Key * parentKey = keyNew ("user:/tests/csvstorage", KEY_VALUE, srcdir_file (file), KEY_END);
+	KeySet * conf = ksNew (20, keyNew ("system:/delimiter", KEY_VALUE, ";", KEY_END),
+			       keyNew ("system:/header", KEY_VALUE, "colname", KEY_END), KS_END);
+	KeySet * ks = ksNew (1, keyNew ("user:/tests/csvstorage", KEY_VALUE, "asdf", KEY_META, "asdf", "asdf", KEY_END), KS_END);
+	PLUGIN_OPEN ("csvstorage");
+	succeed_if (plugin->kdbSet (plugin, ks, parentKey) == ELEKTRA_PLUGIN_STATUS_ERROR,
+		    "kdbSet did not error on invalid meta key insertion");
+	ksDel (conf);
```
Comment (flo91) [inline src/plugins/csvstorage/testmod_csvstorage.c:96]: Sorry, your solution was correct!

You used the macro `PLUGIN_OPEN(NAME)` which calls the function `elektraPluginOpen(...)` with the KeySet `conf` as argument.
https://github.com/ElektraInitiative/libelektra/blob/6a6d1bf57cc50348cbfbe9c94da7dc39c67969ad/tests/cframework/tests_plugin.h#L14-L22

There, the KeySet is assigned to an internal variable:
https://github.com/ElektraInitiative/libelektra/blob/6a6d1bf57cc50348cbfbe9c94da7dc39c67969ad/src/libs/elektra/plugin.c#L92

At the end of your test, the `PLUGIN_CLOSE()` macro is called and there `ksDel (handle->conf)` is done:

https://github.com/ElektraInitiative/libelektra/blob/6a6d1bf57cc50348cbfbe9c94da7dc39c67969ad/src/libs/elektra/plugin.c#L135


```suggestion
```

**Review by flo91 — CHANGES_REQUESTED** [review]

**Review by flo91 — APPROVED** [review]

**Pull request merged**